odl/space/npy_tensors.py

```diff
@@ -0,0 +1,2405 @@
+        >>> x.imag = [2, 3, 4]
+        >>> x
+        cn(3).element([(1+2j), (2+3j), (3+4j)])
+        """
+        if self.space.is_real_space:
+            raise ValueError('cannot set imaginary part in real spaces')
+        self.imag.data[:] = newimag
+
+    def conj(self, out=None):
+        """Return the complex conjugate of ``self``.
+
+        Parameters
+        ----------
+        out : `NumpyTensor`, optional
+            Element to which the complex conjugate is written.
+            Must be an element of ``self.space``.
+
+        Returns
+        -------
+        out : `NumpyTensor`
+            The complex conjugate element. If ``out`` was provided,
+            the returned object is a reference to it.
+
+        Examples
+        --------
+        >>> space = odl.cn(3)
+        >>> x = space.element([1 + 1j, 2, 3 - 3j])
+        >>> x.conj()
+        cn(3).element([ 1.-1.j,  2.-0.j,  3.+3.j])
+        >>> out = space.element()
+        >>> result = x.conj(out=out)
+        >>> result
+        cn(3).element([ 1.-1.j,  2.-0.j,  3.+3.j])
+        >>> result is out
+        True
+
+        In-place conjugation:
+
+        >>> result = x.conj(out=x)
+        >>> x
+        cn(3).element([ 1.-1.j,  2.-0.j,  3.+3.j])
+        >>> result is x
+        True
+        """
+        if self.space.is_real_space:
+            if out is None:
+                return self
+            else:
+                out[:] = self
+                return out
+
+        if not is_numeric_dtype(self.space.dtype):
+            raise NotImplementedError('`conj` not defined for non-numeric '
+                                      'dtype {}'.format(self.dtype))
+
+        if out is None:
+            return self.space.element(self.data.conj())
+        else:
+            if out not in self.space:
+                raise LinearSpaceTypeError('`out` {!r} not in space {!r}'
+                                           ''.format(out, self.space))
+            self.data.conj(out.data)
+            return out
+
+    def __ipow__(self, other):
+        """Return ``self **= other``."""
+        try:
+            if other == int(other):
+                return super(NumpyTensor, self).__ipow__(other)
+        except TypeError:
+            pass
+
+        np.power(self.data, other, out=self.data)
+        return self
+
+    def __int__(self):
+        """Return ``int(self)``."""
+        return int(self.data)
+
+    def __long__(self):
+        """Return ``long(self)``.
+
+        This method is only available in Python 2.
+        """
+        return long(self.data)
+
+    def __float__(self):
+        """Return ``float(self)``."""
+        return float(self.data)
+
+    def __complex__(self):
+        """Return ``complex(self)``."""
+        if self.size != 1:
+            raise TypeError('only size-1 tensors can be converted to '
+                            'Python scalars')
+        return complex(self.data[(0,) * self.ndim])
```

Comment:
The indexing operation takes out the 0-th element and then applies `complex`. I think that will work also with old Numpy since the result is a Numpy scalar.
Perhaps a simpler way to write this is `complex(self.data.ravel()[0])`.